### PR TITLE
matplotlib-base plus bugfix backport

### PR DESCRIPTION
**Pull request by Anthchirp**
Includes everything from #25 (which will automatically be marked as merged on merging this)

Failing build was caused by a newer version of numpy giving slightly different error messages.
Changes were only needed in the tests.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Anthchirp** [comment]
@conda-forge-admin, please rerender

**Comment by Anthchirp** [comment]
Looking good. Just to make sure I didn't inadvertently introduce something:
@conda-forge-admin, please rerender

**Comment by Anthchirp** [comment]
@conda-forge/scikit-bio Ready to review/merge 👍

**Comment by ElDeveloper** [comment]
Thanks so much @Anthchirp, are the main changes mostly focused on the patch to make the tests pass? And how should these be handled once we publish a new release of scikit-bio?

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 1]
```diff
@@ -6,23 +6,28 @@ jobs:
...
-        UPLOAD_PACKAGES: True
+      linux_64_python3.8.____cpython:
+        CONFIG: linux_64_python3.8.____cpython
+        UPLOAD_PACKAGES: 'True'
         DOCKER_IMAGE: condaforge/linux-anvil-comp7
+  timeoutInMinutes: 360
+
   steps:
+  - script: |
+         rm -rf /opt/ghc
```
Comment (ElDeveloper) [inline .azure-pipelines/azure-pipelines-linux.yml:27]: This seems a little odd, do you happen to know what's going on here?

**Review by ElDeveloper — APPROVED** [review]
Thanks so much!

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 1 above)
Comment (Anthchirp) [reply .azure-pipelines/azure-pipelines-linux.yml:27]: All of those files are maintained by the rerendering process. I have no idea what the reasons behind the individual bits and pieces are.

**Comment by Anthchirp** [comment]
> Thanks so much @Anthchirp

you're welcome

> are the main changes mostly focused on the patch to make the tests pass? And how should these be handled once we publish a new release of scikit-bio?

Yes, only the tests needed fixing. Once you make a new release you (or the auto-update bot) will create in a pull request here to build the updated package. Since the patches will no longer apply cleanly that pull request should break. At that stage all you need to do is remove the patches and delete the associated `.patch` file.

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 1 above)
Comment (Anthchirp) [reply .azure-pipelines/azure-pipelines-linux.yml:27]: This is the story behind that line:
https://github.com/conda-forge/conda-smithy/commit/de02148e179bca2b8f249eb96d523da1c1958642
and
https://github.com/conda-forge/conda-smithy/pull/1345

Comment (ElDeveloper) [reply .azure-pipelines/azure-pipelines-linux.yml:27]: Thanks for the pointers!

**Comment by ElDeveloper** [comment]
Great, thanks @Anthchirp.

@ebolyen or @thermokarst would anyone of you be able to review?

**Comment by thermokarst** [comment]
Thanks @Anthchirp & @ElDeveloper - I'll take a look at this today.

**Review by thermokarst — APPROVED** [review]
LGTM, I say we merge this and see what happens. There is a _ton_ of noise in the changeset, but the core of the changes looks pretty reasonable to me. Thanks for putting this together @Anthchirp!

**Comment by ElDeveloper** [comment]
Thanks so much @Anthchirp & @thermokarst

**Comment by thermokarst** [comment]
Looks like the build failed, @Anthchirp. I don't have any more availability today to look into this, but will circle back in a few days.

**Comment by Anthchirp** [comment]
I've asked in gitter and the suggestion was to rerender once more and try again. I'll open a PR for that

**Pull request merged**